chain/types/ethtypes/eth_types.go

```diff
@@ -1025,6 +1025,24 @@ type EthTraceTransaction struct {
 	TransactionPosition int     `json:"transactionPosition"`
 }
 
+type EthTraceFilterResult struct {
+	*EthTrace
+	BlockHash           EthHash `json:"blockHash"`
+	BlockNumber         int64   `json:"blockNumber"`
+	TransactionHash     EthHash `json:"transactionHash"`
+	TransactionPosition int     `json:"transactionPosition"`
+}
+
+// EthTraceFilterCriteria defines the criteria for filtering traces.
+type EthTraceFilterCriteria struct {
+	FromBlock   string   `json:"fromBlock,omitempty"`
+	ToBlock     string   `json:"toBlock,omitempty"`
+	FromAddress []string `json:"fromAddress,omitempty"`
+	ToAddress   []string `json:"toAddress,omitempty"`
+	After       int      `json:"after,omitempty"`
+	Count       int      `json:"count,omitempty"`
```

Comment:
I think you should make these all pointers since they are properly optional and you can't distinguish an empty value from an omitted value. `EthFilterSpec` is the closest relation to this I think. You'll just have to pay very careful attention when checking them and dereferencing.

Reply:
That's a good idea, thanks! I appreciate you explaining why pointers solve the problem of distinguishing empty values from omitted values.

Reply:
f74fd6001